test/test_sparse.py

```diff
@@ -2364,10 +2398,10 @@ def test_factory_device_type_inference(self):
                 self.assertEqual(should_be_cuda, t.is_cuda)
                 self.assertEqual(t.is_cuda, t_empty.is_cuda)
 
-    @cpu_only
-    def test_factory_copy(self):
+    @onlyCPU
+    def test_factory_copy(self, device):
```

Comment:
why aren't we running this test for complex types?

Reply:
I added test cases for complex data